File: caves-of-zircon.java/src/main/java/rltut/views/LoseView.java
package rltut.views;

import org.hexworks.zircon.api.grid.TileGrid;
import org.hexworks.zircon.api.input.Input;
import org.hexworks.zircon.api.input.InputType;
import rltut.ai.Creature;

public class LoseView implements View {
    private Creature player;
    private TileGrid tileGrid;

    public LoseView(TileGrid tileGrid, Creature player) {
        this.player = player;
        this.tileGrid = tileGrid;
    }

    @Override
    public void dock() {
        // TODO: use dialogs
//        tileGrid.writeCenter("R.I.P.", 3);
//        tileGrid.writeCenter(player.causeOfDeath(), 5);
//        tileGrid.writeCenter("-- press [enter] to restart --", 22);
    }

    @Override
    public View respondToUserInput(Input input) {
        return input.inputTypeIs(InputType.Enter) ? new PlayView(tileGrid) : this;
    }
}

File: caves-of-zircon.kotlin/src/main/kotlin/org/hexworks/cavesofzircon/view/LoseView.kt
package org.hexworks.cavesofzircon.view

import org.hexworks.zircon.api.Components
import org.hexworks.zircon.api.component.ComponentAlignment
import org.hexworks.zircon.api.component.ComponentAlignment.*
import org.hexworks.zircon.api.graphics.BoxType
import org.hexworks.zircon.api.grid.TileGrid
import org.hexworks.zircon.api.kotlin.onMouseReleased

class LoseView(tileGrid: TileGrid) : BaseView(tileGrid) {

    init {
        val msg = "You lost!"
        val header = Components.textBox()
                .withContentWidth(msg.length)
                .addHeader(msg)
                .addNewLine()
                .withAlignmentWithin(tileGrid, ComponentAlignment.CENTER)
                .build()
        val restartButton = Components.button()
                .withAlignmentAround(header, BOTTOM_LEFT)
                .withText("Restart")
                .wrapSides(false)
                .wrapWithBox()
                .withBoxType(BoxType.SINGLE)
                .build()
        val exitButton = Components.button()
                .withAlignmentAround(header, BOTTOM_RIGHT)
                .withText("Quit")
                .wrapSides(false)
                .wrapWithBox()
                .withBoxType(BoxType.SINGLE)
                .build()

        restartButton.onMouseReleased {
            PlayView(tileGrid).dock()
        }

        exitButton.onMouseReleased {
            System.exit(0)
        }

        screen.addComponent(header)
        screen.addComponent(restartButton)
        screen.addComponent(exitButton)
    }
}
